### PR TITLE
Update JFR image

### PR DESCRIPTION
**Pull request by anbrsap**
### Description

- Update the JFR image to tag `231006_5085985`
- Change Helm chart parameters

### Submitter checklist

- [x] Change has been tested (on a back-end cluster)
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [x] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [x] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [x] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There is at least 1 approval for the pull request and no outstanding requests for change
- [x] All voter checks have passed
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] The Pull Request title is understandable and reflects the changes well
- [x] The Pull Request description is understandable and well documented
- [x] [changelog.yaml] entry for this Pull Request has been added
    - [x] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: changelog.yaml [hunk 1]
```diff
@@ -52,6 +52,28 @@
...
+        - If needed, set `pipelineRuns.logging.forwarder.emitTimeoutMillis` (`emitMaxRetriesIfBufferFull` was removed)
+        - If needed, set `pipelineRuns.logging.forwarder.useSidecar` to `true` (default value changed to `false`)
+        - If needed, set `pipelineRuns.logging.forwarder.tag` (default value changed to `logs`)
+
+        See the chart README for details on chart parameters.
+      pullRequestNumber: 408
+      jiraIssueNumber: 1258
+
 - version: "0.37.0"
   date: 2023-09-01
```
Comment (valiparsa) [inline changelog.yaml:78]:
```suggestion
  date: 2023-09-01
  recall:
    message: |-
      elasticsearch-logs-plugin in this version contained some bugs which are fixed in the next patch version. Please use version 0.37.1 instead.
```

File: charts/steward/values.yaml [hunk 2]
```diff
@@ -48,30 +48,31 @@ pipelineRuns:
...
+      maxWaitSecondsUntilFlusherTerminated: ~
+      bufferChunkInitialSize: ~
+      bufferChunkRetentionSize: ~
+      bufferChunkRetentionTimeMillis: ~
+      flushAttemptIntervalMillis: ~
+      maxBufferSize: ~
+      emitTimeoutMillis: ~
   jenkinsfileRunner:
-    image: "stewardci/stewardci-jenkinsfile-runner:230825_1c09345"
+    image: "stewardci/stewardci-jenkinsfile-runner:TEST_231002_8e67897"
```
Comment (valiparsa) [inline charts/steward/values.yaml:75]: I'll comment here just as a reminder to update the tag once JFR is released.

**Review by valiparsa — APPROVED** [review]
The proposed change looks good to me

File: charts/steward/values.yaml (hunk 2 above)
Comment (anbrsap) [reply charts/steward/values.yaml]: Set to `231006_5085985`.

**Pull request merged**